casanovo/denovo/model.py

```
@@ -1293,11 +1308,19 @@ def _psm_batches(
         for i, (precursor_charge, precursor_mz) in enumerate(
             zip(charge_iter, mz_iter)
         ):
-            for cand in self.protein_database.get_candidates(
+            spec_cands = self.protein_database.get_candidates(
                 precursor_mz, precursor_charge
-            ):
+            )
+            for cand in spec_cands:
                 candidates.append((i, cand))
```

Ruff (0.15.6)

[warning] Use `list.extend` to create a transformed list

(PERF401)